rcl/src/rcl/wait.c

```diff
@@ -267,53 +210,47 @@ rcl_wait_set_get_allocator(const rcl_wait_set_t * wait_set, rcl_allocator_t * al
   wait_set->impl->RMWCount++;
 
 #define SET_CLEAR(Type) \
-  RCL_CHECK_ARGUMENT_FOR_NULL(wait_set, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator()); \
-  if (!__wait_set_is_valid(wait_set)) { \
-    RCL_SET_ERROR_MSG("wait set is invalid", rcl_get_default_allocator()); \
-    return RCL_RET_WAIT_SET_INVALID; \
-  } \
-  memset( \
-    (void *)wait_set->Type ## s, \
-    0, \
-    sizeof(rcl_ ## Type ## _t *) * wait_set->size_of_ ## Type ## s); \
-  wait_set->impl->Type ## _index = 0;
+  do { \
+    memset( \
+      (void *)wait_set->Type ## s, \
+      0, \
+      sizeof(rcl_ ## Type ## _t *) * wait_set->size_of_ ## Type ## s); \
+    wait_set->impl->Type ## _index = 0; \
+  } while (false)
```

Comment:
Why is the `do` / `while (false)` necessary here?

Same for the other macros.

Reply:
They're not necessary. I changed it to make it require a semicolon because I remember the last discussion about ros2/rcutils#113 seemed to favor that direction.